### PR TITLE
Graph visualization

### PR DESCRIPTION
**Pull request by rajcscw**

### CODE REVIEW

**Comment by rajcscw** [comment]
Nice work Lars, I will keep checking the changes until tomorrow since they are a lot :)

**Comment by LarsHill** [comment]
Thanks @rajcscw No worries. Before we can merge there are some things missing. 

- We have to adapt the readme and examples to the api changes.
- Also, I want to add some improvements wrt paging the console graph visualization.
- Some commenting cleanup is still needed.

File: fluidml/common/logging.py [hunk 1]
```diff
@@ -8,26 +8,41 @@
...
-    tb_lib = True
-except ImportError:
-    tb_lib = False
+from rich.console import Console
+from rich.logging import RichHandler
+from tblib import pickling_support
+
+
+console = Console(stderr=True)
+logging.lastResort = RichHandler(console=console,
```
[the remark this answers is not in the document]
Comment (LarsHill) [reply fluidml/common/logging.py:15]: Haha this is the logging API ;) Seems, they implemented it at a time prior to pep-8 ;)

File: fluidml/flow/ascii_graph_visualization.py [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+        )
+
+        canvas.text(
+            int(round(x - minx)) + 1, int(round(y - miny)) + 1, vertex.data
+        )
+
+    return canvas.draw()
+
+
+# import grandalf
```
[the remark this answers is not in the document]
Comment (LarsHill) [reply fluidml/flow/ascii_graph_visualization.py]: Btw, this is removed already I think :)

File: README.md [hunk 3]
```diff
@@ -167,6 +167,9 @@ class EvaluateTask(Task):
 Next, we can create the defined tasks with their specifications. We now only write their specifications, later these are used to create real instances of tasks by FluidML.
 For each Task specification, we also add a list of result names that the corresponding task _publishes_ and _expects_. Each published result object will be considered when results are automatically collected for a successor task.
 
+Note: The `task_kwargs` argument holds the configuration of the task. It has to be a dictionary (possibly nested), which is `json` serializable. 
+That means a `TypeError` is thrown if the dictionary contains objects, e.g. an `np.array`, that cannot be serialized by Python's json encoder.
```
[the remark this answers is not in the document]
Comment (LarsHill) [reply README.md:174]: As discussed, let's o with **Solution A** for now. An additional argument `additional_kwargs` is available to `TaskSpec` and `GridTaskSpec` to provide arbitrary python objects to tasks. They are ignored for our redundancy checks/config comparisons. 
ALso, I renamed `task_kwargs` to `config` to set it apart from `additional_kwargs`.

File: fluidml/flow/flow.py [hunk 4]
```diff
@@ -21,68 +32,117 @@ class Flow:
...
+        # contains the original, user defined task spec graph as networkx DiGraph
+        # -> accessible to the user for introspection and visualization
+        self.task_spec_graph: Optional[DiGraph] = None
+
+    @staticmethod
+    def _check_acyclic(graph: DiGraph) -> None:
+
+        # try to find cycles in graph -> return None if None are found
+        try:
+            edges = nx.find_cycle(graph, orientation='original')
```
[the remark this answers is not in the document]
Comment (LarsHill) [reply fluidml/flow/flow.py:56]: I checked the API and the function you're suggesting doesn't detect cycles containing of more than three edges. For now, I would stick to the current implementation.

**Pull request merged**